rest/changes_api_test.go

```diff
@@ -763,21 +763,36 @@ func TestChangeWaiterExitOnChangesTermination(t *testing.T) {
 			require.NoError(t, err)
 
 			lastSeq := c.Last_Seq.(string)
-			t.Logf("lastSeq: %v %v", lastSeq, c)
 
-			startGoroutines := runtime.NumGoroutine()
-			t.Logf("start goroutines: %d", startGoroutines)
+			assert.NoError(t, rt.WaitForCondition(func() bool {
+				return rt.GetDatabase().DbStats.CBLReplicationPull().NumPullReplCaughtUp.Value() == 0
+			}))
 
 			// Send all of these concurrently, just to speed up testing. Not reliant on concurrency to trigger goroutine leak.
 			wg := sync.WaitGroup{}
 			for i := 0; i < numChangesToSend; i++ {
 				wg.Add(1)
 				go func() {
-					resp := sendRequestFn(rt, test.username, http.MethodGet, "/db/_changes?since="+lastSeq+"&feed="+test.feedType+"&timeout=100", "")
+					resp := sendRequestFn(rt, test.username, http.MethodGet, "/db/_changes?since="+lastSeq+"&feed="+test.feedType+"&timeout="+requestTimeoutMs, "")
 					assertStatus(t, resp, http.StatusOK)
 					wg.Done()
 				}()
 			}
+
+			// normal does not wait for any further changes, so does not have any replications in a "caught up" state.
+			if test.feedType != "normal" {
+				assert.NoError(t, rt.WaitForCondition(func() bool {
+					// can't check for equality with numChangesToSend because they might've timed out by now on slow test runs.
+					// Assume there's at least one still running though.
+					return rt.GetDatabase().DbStats.CBLReplicationPull().NumPullReplCaughtUp.Value() > 0
+				}))
+			} else {
+				assert.NoError(t, rt.WaitForCondition(func() bool {
```

Comment:
Can be removed

Reply:
I've pushed a change with this removed, and a new "total" stat for caught up. This has allowed me to simplify the test by removing the async changes request, and shortening the 2.5s timeout.